geopandas/geodataframe.py

```diff
@@ -1673,7 +1679,26 @@ def dissolve(
 
         # Process non-spatial component
         data = self.drop(labels=self.geometry.name, axis=1)
-        aggregated_data = data.groupby(**groupby_kwargs).agg(aggfunc)
+        with warnings.catch_warnings(record=True) as record:
+            aggregated_data = data.groupby(**groupby_kwargs).agg(aggfunc, **kwargs)
+        for w in record:
+            if str(w.message).startswith("The default value of numeric_only"):
+                msg = (
+                    f"The default value of numeric_only in aggfunc='{aggfunc}' "
+                    "within pandas.DataFrameGroupBy.agg used in dissolve is "
+                    "deprecated. In pandas 2.0, numeric_only will default to False. "
+                    "Either specify numeric_only or select only columns which should "
+                    "be valid for the function."
```

Comment:
```suggestion
                    "Either specify numeric_only as additional argument in dissolve() "
                    "or select only columns which should be valid for the function."
```

I was wondering if it is clear enough from the current warning that `numeric_only=True` can be specified in dissolve itself, since it is actually a keyword of the groupby method in pandas